sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/InstallerUtils.java

```diff
@@ -121,4 +122,79 @@ public static boolean isUnknownSourcesEnabled(@NonNull Context context) {
             return INSTALL_NON_MARKET_APPS_ENABLED.equals(Settings.Secure.getString(context.getContentResolver(), Settings.Secure.INSTALL_NON_MARKET_APPS));
         }
     }
+
+    /**
+     * Check whether user enabled start activity from the background.
+     *
+     * @param context any context.
+     * @return true if start activity from the background is enabled, false otherwise.
+     */
+    public synchronized static boolean isSystemAlertWindowsEnabled(@NonNull Context context) {
+
+        /*
+        * From Android 10 (29 API level) or higher we have to
+        * use this permission for restarting activity after update.
+        * See more about restrictions on starting activities from the background:
+        * - https://developer.android.com/guide/components/activities/background-starts
+        * - https://developer.android.com/about/versions/10/behavior-changes-all#sysalert-go
+        */
+        return Build.VERSION.SDK_INT < Build.VERSION_CODES.Q ||
```

Comment:
Sorry for the unsolicited comment, but isn't it relevant only for Android Go (Low ram) devices?
Instead of all `>= Build.VERSION_CODES.Q`

Reply:
See https://stackoverflow.com/a/65176398

Reply:
Hi! Thanks for your suggestion!
Actually, I'm not sure. The public docs [describe](https://developer.android.com/about/versions/10/behavior-changes-all#sysalert-go):
> Apps running on Android 10 (Go edition) devices cannot receive the SYSTEM_ALERT_WINDOW permission.

Your link suggests using `isLowRamDevice` API to check this permission, but in the [public docs](https://developer.android.com/reference/android/app/ActivityManager#isLowRamDevice())`isLowRamDevice` API has a different purpose.
Unfortunately, I do not have the required device to make sure that this solution will work on these devices and that result of this API doesn't depend on some other permissions.

Reply:
I think they use `isLowRamDevice` to detect the Android Go devices, which are low end devices.
https://www.android.com/versions/go-edition/
"A lightning speed OS, built for smartphones with 2 GB of RAM or less."

They use this because "SYSTEM_ALERT_WINDOW is disabled on on low ram devices starting from Q"

I was worried the PR would make us ask an unnecessary permission on non-Android Go device and got confused.
Looks like worst case you will have an unexpected behaviour on the Android Go devices.

Anyway, you have much more context than me 😅 
Thanks for looking at my comment. 🙌

Reply:
We use this permission only on Androids 10 and higher for the ability to restart the application after updating. This permission is optional for the users and can be ignored. In the worst case, it just won't open the app after the update.
Thanks for your attention and thoughts on this PR :).